VERSIONING_POLICY.md

```diff
@@ -0,0 +1,33 @@
+# Open OnDemand Versioning Policy
+## Version Numbering
+Open OnDemand code releases utilize a XXX.YYY.ZZZ numbering scheme (e.g. 2.0.29), where:
+* XXX is the major version number
+* YYY is the minor version number
+* ZZZ is the patch version number
+
+## Major Versions
+New major version numbers will typically be implemented for any of the following reasons:
+* Significant breaking changes that prevent seamless upgrading from previous versions
+* Significant deprecations related to software dependencies
```

Comment:
Deprecations should not require a major bump.  Deprecations generally mean that something will produce a warning or otherwise documented as "don't use this" but it will still work.  Removing a previously deprecated thing would be where you do major bump.

Reply:
Maybe I'm using the wrong word then here?  Open to suggestions on how to better phrase this.  For example, upgrading the required Python from 2.7 to 3.0 (which isn't fully backward compatible with 2.7) is a 'deprecation' in my mind that would trigger a major version.

Reply:
Deprecation to me is when something is no longer supported but still works, ie we say "don't use this feature, because we are getting rid of it in the next release, but it still works".  So I think deprecation is just the wrong word here.  I think deprecations would be better for minor releases and the removal of a previously deprecated feature is something for major release.

WRT to Python 2.7 to 3.0 I think maybe wording it as "Changes to OnDemand dependencies that are not backwards compatible, eg: require Python 3.0 instead of 2.7".  Maybe another bullet would be "Removal of previously deprecated features".  Then for minor release a bullet would be "Deprecation of previously supported features".

I think where this gets a little slippery is Ruby and NodeJS reach EOL a lot faster than Python in RHEL so we will have to be updating our dependencies possibly more frequently than the 1-2 year cadence for major releases.

Reply:
I've removed that and added a bit about dependencies. Specifically that dependency updates can come at any time in any type of release.